core/src/main/java/org/apache/accumulo/core/iterators/user/VisibilityFilter.java

```diff
@@ -73,6 +73,15 @@ public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> op
     this.cache = new LRUMap<>(1000);
   }
 
+  @Override
+  public SortedKeyValueIterator<Key,Value> deepCopy(IteratorEnvironment env) {
+    VisibilityFilter result = (VisibilityFilter) super.deepCopy(env);
+    result.filterInvalid = this.filterInvalid;
+    result.accessEvaluator = this.accessEvaluator;
+    result.cache = new LRUMap<>(1000);
```

Comment:
I wonder if it shouldn't make a new cache:

```suggestion
    result.cache = this.cache;
```

Reply:
Agreed, no reason to create a new one.